### ISSUE TITLE
Improve assets search

### ISSUE THREAD

**Issue by nebolax** [rotki/rotki#4936]
## Abstract

Currently we have 2 different approaches for assets search. Levenshtein distance and search by a substring.

The problem is that:
1. Right now we are doing levenshtein in python (not on sqlite level), which means that if only one page of assets is requested, we still have to load all the assets into memory to search.
2. Search by substring is very light but is not really accurate since you can't really make sure that, for example, "DAI" always appears before "aDAI".

## Motivation

Improve search quality without making it too heavy.

## Specification

Ideas:
- Use sqlite's levenshtein plugin for all searches, so that we don't have to load assets into memory.
- If first doesn't work, maybe keep levenshtein in python, but try to reduce required memory as much as possible.

**Comment by LefterisJP** [comment]
Is this actionable @nebolax? Or is it just ideas which we probably won't follow? If latter let's just close.

**Comment by skylarweaver** [comment]
It would be great if the search was improved such that you could search for a 0x address and have it show you the respective wallet name that you called it.

Having a 0x address and then trying to find which one it is in your rotki portfolio is actually quite difficult.

**Comment by LefterisJP** [comment]
> It would be great if the search was improved such that you could search for a 0x address and have it show you the respective wallet name that you called it.
> 
> Having a 0x address and then trying to find which one it is in your rotki portfolio is actually quite difficult.
> 
> Not sure if this is the right issue or if this deserves a separate one

Hey Skylar, this sounds different. Are you talking about accounts or about tokens/assets?

Can you describe the problem in a bit more detail? Or let me try to see if I understand what you are talking about:

The problem: Let's say that you own `0x01B53d796C8b55ef29E022C01e5f1c6C849120a5` and you called it: `My ledger Nano X wallet` as the name. After a while, as you gather lots and lots of tracked addreses you forget what name you gave to each account. You can easily see the names of all your accounts in the table when you go to the "Accounts & Balances" view, but if you just have the `0x01B53d796C8b55ef29E022C01e5f1c6C849120a5` address there is no way to see which account this is?

Do I understand correctly? So you want to be able to better search your tracked addresses?

**Comment by skylarweaver** [comment]
Yes this is exactly the issue! I will create a new issue. And I will borrow your description of it as it's more detailed than mine 😅

**Comment by LefterisJP** [comment]
This is not trivial.

I don't think it comes with any sqlite version. It's actually just a module which you need to fetch, compile (and best do so with `SQLITE_ENABLE_ICU` for non ascii comparison too),ship with the app and also load as an extension whenever we open the DB.

Would be a nice improvement but it's not trivial
